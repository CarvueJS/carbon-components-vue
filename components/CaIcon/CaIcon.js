import icons from 'carbon-icons';

export default {
  name: 'ca-icon',
  data: () => ({
    iconElement: null,
  }),
  props: {
    description: {
      type: String,
      default: '',
    },
    iconName: {
      type: String,
      default: '',
    },
  },
  methods: {
    getIcon(name, iconsObj = icons) {
      const icon = iconsObj.filter(obj => obj.name === name);

      if (icon.length === 0) {
        return false;
      } else if (icon.length > 1) {
        throw new Error('[Carvue]: Multiple icons found');
      }
      const iconSvg = icon[0].svgData;

      return Object.keys(iconSvg).filter(key => iconSvg[key]).map((key) => {
        const data = iconSvg[key];
        if (key === 'circles') {
          return data.map((circle, index) => {
            const circleProps = {
              cx: circle.cx,
              cy: circle.cy,
              r: circle.r,
              key: `circle${index}`,
            };
            return <circle {...circleProps} />;
          });
        } else if (key === 'paths') {
          return data.map((path, index) => {
            const pathProps = {
              d: path.d,
              key: `key${index}`,
            };
            return <path d={pathProps.d} key={pathProps.key} />;
          });
        }
        return '';
      });
    },
  },
  render() {
    const svgContent = this.getIcon(this.iconName);

    return (
      <svg>
        <title>{this.description}</title>
        {svgContent}
      </svg>
    );
  },
};
