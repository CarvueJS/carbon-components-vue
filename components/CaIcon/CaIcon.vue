<template>
  <svg>
    <title class='iconTitle'>{{description}}</title>
  </svg>
</template>

<script>
import icons from 'carbon-icons'

export default {
  name: 'ca-icon',
  data: () => ({
    iconElement: null
  }),
  props: {
    description: {
      type: String,
      default: ''
    },
    iconName: {
      type: String,
      default: ''
    }
  },
  mounted () {
    // console.log(this.getIcon(this.iconName)[0][0])
    // var path = document.createElement("path");
    // path.setAttribute('d', this.getIcon(this.iconName)[0][0].d);
    // path.setAttribute('key', this.getIcon(this.iconName)[0][0].key);
    // this.$el.querySelector('.iconTitle').append(path)
  },
  methods: {
    getIcon(name, iconsObj = icons) {
      const icon = iconsObj.filter(obj => obj.name === name);

      if (icon.length === 0) {
        return false;
      } else if (icon.length > 1) {
        throw new Error('[Carvue]: Multiple icons found')
      } 
      const iconSvg = icon[0].svgData;

      const svgElement = Object.keys(iconSvg).filter(key => iconSvg[key]).map(key => {
        const data = iconSvg [key];
        if (key === 'circles') {
          return data.map((circle, index) => {
            const circleProps = {
              cx: circle.cx,
              cy: circle.cy,
              r: circle.r,
              key: `circle${index}`
            }
            return `<circle cx=${circleProps.cx}
                            cy=${circleProps.cy}
                            r=${circleProps.r}
                            key=${circleProps.key}/>`;
          })
        } else if (key === 'paths') {
          return data.map((path, index) => {
            const pathProps = {
              d: path.d,
              key: `key${index}`
            }
            return pathProps
          })
        } else {
          return ''
        }
      })
      return svgElement
    },

  }
}
</script>

