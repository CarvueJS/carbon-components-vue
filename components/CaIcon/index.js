import CaIcon from './CaIcon';

CaIcon.install = (Vue) => {
  Vue.component(CaIcon.name, CaIcon);
};

export default CaIcon;
