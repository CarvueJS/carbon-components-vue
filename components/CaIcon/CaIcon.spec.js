import { shallow } from 'vue-test-utils';
import CaIcon from './CaIcon.vue';

describe('CaIcon', () => {
  test('is a Vue instance', () => {
    const wrapper = shallow(CaIcon);
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
})
