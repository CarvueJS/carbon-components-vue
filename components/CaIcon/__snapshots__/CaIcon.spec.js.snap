// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaIcon Mathes snapshot 1`] = `
<svg width="16" height="16" viewBox="0 0 25 25" fill-rule="evenodd" class="">
  <title></title>
</svg>
`;
